### ISSUE TITLE
Results in NatMamba+ vs NetMamba papers

### ISSUE THREAD

**Issue by gayanku** [wangtz19/NetMamba#27]
Could you pls explain what the reported results in your NatMamba+ paper for NetMamba model differs from the reported results in the earlier NetMamba paper?  Is the datasets different?

Results in the NetMamba paper
<img width="719" height="207" alt="Image" src="https://github.com/user-attachments/assets/85b69126-5cc2-4016-8664-ba2480310fd6" />

Results in the NetMamba+ paper
<img width="667" height="269" alt="Image" src="https://github.com/user-attachments/assets/5d256a4a-a2bd-41be-8735-3e458a6f5775" />

**Comment by mann1x** [comment]
@gayanku 

Guess it's due to the different training (120 vs 400 epochs) and pre-processing.
In the original paper I don't see it specified but for NetMamba+ they used MFC=2000 cap and 8:1:1 splits.

I can't replicate the same from the authors but my own NetMamba "v4" encoder, fine-tuned loosely as in the paper, is giving results more similar to those in the NetMamba+ papers.

```
  ┌─────────┬────────┬─────────────────────┬─────────────────────────┐
  │ Dataset │   v4   │ NetMamba (original) │ NetMamba (in NM+ paper) │
  ├─────────┼────────┼─────────────────────┼─────────────────────────┤
  │ USTC    │ 98.59% │ 99.90%              │ 99.57%                  │
  ├─────────┼────────┼─────────────────────┼─────────────────────────┤
  │ ISCXVPN │ 92.22% │ 96.21%              │ 94.90%                  │
  └─────────┴────────┴─────────────────────┴─────────────────────────┘
```

It's not a really one to one comparison as I did have only some flows from the datasets:
  1. Only 14/20 USTC classes (missing 6), 228K
  2. Encoder was pre-trained on security/IoT traffic (5.1M flows), not the paper's Browser+Kitsune corpus (317K flows)
  3. ISCXVPN only 257 test samples, high variance

The model is incredibly strong, my v4 was trained on CIC-IDS2017+IoT-23+CTU-13.
Achieving 98.6% on USTC is terrific. On security achieves 99.55% accuracy.

**Comment by RyanLi888** [comment]
@mann1x   Has the code and weights of NetMamba+ been released?Where did you implement your reproduction, and did you re-pre-train the model from scratch yourself?

**Comment by mann1x** [comment]
@RyanLi888 
No but the paper has enough information to make your own. It's basically the same NetMamba adding the multimodal for IAT.
That above, v4, is the old NetMamba architecture, without IAT.

I'm pre-training and fine-tuning my NetMamba Plus (v7) from scratch, starting hopefully today. 
Tried to re-use the NetMamba Phase A pre-training as a base but it was not a good idea.
For NM+ I will do pre-training with 3,2M flows and fine-tuning security on 5M flows.

**Comment by wangtz19** [comment]
Thank you all for your interest in NetMamba+.
The differences in experimental results may be attributed to factors such as varying flow sampling strategies, different random seeds used for training/validation/test splits, and differences in processing header fields (e.g., IP addresses and ports).

Regarding the NetMamba+ code, the manuscript is currently undergoing the second-round revision at a journal. I will make the code publicly available as soon as possible.

**Comment by mann1x** [comment]
@wangtz19 
Thank you! As said, it's an excellent model.

If you have any additional tips for the training of NM+, very welcome, I'd appreciate that.
With the original NM, the biggest issue I had was not using BF16.

I already attempted training NM+ and it was an epic failure; I realized too late that most flows didn't have enough frames, had to rebuild from scratch the corpus.
